browser/net/brave_request_handler.cc

```diff
@@ -100,6 +103,12 @@ void BraveRequestHandler::SetupCallbacks() {
       base::Bind(brave::OnBeforeStartTransaction_SiteHacksWork);
   before_start_transaction_callbacks_.push_back(start_transaction_callback);
 
+  if (base::FeatureList::IsEnabled(features::kGlobalPrivacyControl)) {
```

Comment:
it's better to encapsulate feature checks inside the relevant code, so rather than conditionally including the helper, you should always include it and conditionally add the header inside it